syntax = "proto3";

message ResponseData
{
  string error = 1;
  string output = 2;
}

message RegisterData {
  string email = 1;
  string pwd = 2;
}

message RegisterDataReply {
  ResponseData responseData = 1;
}

message TokenResponse {
  string token = 1;
  uint64 lifeTimeInSec = 2;

}

message Message {
  string reciever = 1;
  string sender = 2;
  string data = 3;
}

service Auth {
  rpc Register (RegisterData) returns (RegisterDataReply) {}
  rpc Login (RegisterData) returns (TokenResponse) {}
  rpc ExtendTokenLifetime (RegisterData) returns (TokenResponse) {}
  rpc SendMsg (Message) returns (ResponseData) {}
}


